trieste/acquisition/multi_objective/scalarization.py

```diff
@@ -0,0 +1,212 @@
+# Copyright 2020 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" This module contains interface for scalarization functions and some example functions. """
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Callable, Mapping
+
+import tensorflow as tf
+
+from trieste.data import Dataset
+from trieste.models.interfaces import (
+    HasTrajectorySampler,
+)
+from trieste.types import Tag, TensorType
+
+
+IdealSpecCallable = Callable[..., TensorType]
```

Comment:
Why not make this precise?

```suggestion
IdealSpecCallable = Callable[[Mapping[Tag, HasTrajectorySampler], datasets: Mapping[Tag, Dataset]], TensorType]
```